spec/lib/disability_compensation/providers/document_upload/lighthouse_supplemental_document_upload_provider_spec.rb

```diff
@@ -58,13 +69,50 @@
 
   describe 'submit_upload_document' do
     let(:faraday_response) { instance_double(Faraday::Response) }
+    let(:lighthouse_request_id) { Faker::Number.number(digits: 8) }
 
-    it 'submits the document via the UploadSupplementalDocumentService and returns the response' do
+    before do
       allow(BenefitsDocuments::Form526::UploadSupplementalDocumentService).to receive(:call)
         .with(file_body, lighthouse_document)
         .and_return(faraday_response)
 
-      expect(provider.submit_upload_document(lighthouse_document, file_body)).to eq(faraday_response)
+      allow(faraday_response).to receive(:body).and_return(
+        {
+          'data' => {
+            'success' => true,
+            'requestId' => lighthouse_request_id
+          }
+        }
+      )
+    end
+
+    it 'uploads the document via the UploadSupplementalDocumentService' do
+      expect(BenefitsDocuments::Form526::UploadSupplementalDocumentService).to receive(:call)
+        .with(file_body, lighthouse_document)
+
+      provider.submit_upload_document(lighthouse_document, file_body)
+    end
+
+    it 'increments a StatsD success metric' do
+      expect(StatsD).to receive(:increment).with(
+        'my_stats_metric_prefix.lighthouse_supplemental_document_upload_provider.success'
+      )
+
+      provider.submit_upload_document(lighthouse_document, file_body)
+    end
+
+    it 'creates a pending Lighthouse526DocumentUpload record for the submission so we can poll Lighthouse later' do
+      upload_attributes = {
+        aasm_state: 'pending',
+        form526_submission_id: submission.id,
+        # Polling record type mapped to L023 used in tests
+        document_type: Lighthouse526DocumentUpload::BDD_INSTRUCTIONS_DOCUMENT_TYPE,
```

Comment:
That code smell I mentioned creating some confusing misdirection in the tests, I think the migration code in general in general has gotten a little unwieldy as you can see in this PR but I bet we can fix that